### PR TITLE
Feature/realm encryption

### LINKED ISSUE #219: Realm add db encryption.
We should add encryption. 

Realm supports encrypting the database file on disk with AES-256+SHA2 by supplying a 64-byte encryption key when creating a Realm.

### PR DESCRIPTION
**Pull request by OlegGordiichuk**
- Generate secure key with help of (SecRandomCopyBytes)
- Save key to keychain
- Had  implement usage of the secure Realm configuration
- Had written pure function to take care of old unencrypted databases.

Fix #219

### CODE REVIEW

**Comment by OlegGordiichuk** [comment]
@hewigovens could you review this pr also please?

File: Trust/Core/Types/TrustRealmConfiguration.swift [hunk 1]
```diff
@@ -3,32 +3,95 @@
...
-                config.fileURL = oldURL
-                return config
+                return Realm.Configuration(fileURL: oldURL)
             }
         }
-        config.fileURL = newURL
-        return config
+
+        guard let key = getKey(with: waletsRealmKey) else {
+            let newKey = PasswordGenerator.generateRandomData(bytesCount: 64)
```
Comment (hewigovens) [inline Trust/Core/Types/TrustRealmConfiguration.swift]: Similar code in sharedConfiguration, can we reuse them?

File: Trust/Core/Types/TrustRealmConfiguration.swift [hunk 2]
```diff
@@ -3,32 +3,95 @@
...
+            _ = realmDatabases.map { encryptDatabase(with: $0, and: encryptionKey, and: schemaVersion) }
+        } catch {
+            print(error.localizedDescription)
+        }
+    }
+
+    private static func encryptDatabase(with path: URL, and encryptionKey: Data, and schemaVersion: UInt64) {
+        let documentsPath = path.deletingLastPathComponent()
+        let databaseUrls = [path,
+            path.appendingPathExtension("lock"),
```
Comment (hewigovens) [inline Trust/Core/Types/TrustRealmConfiguration.swift]: It's safe to remove these files?

**Comment by hewigovens** [comment]
If user delete the app and install it again, should we use the same encryption key?

File: Trust/Core/Types/TrustRealmConfiguration.swift [hunk 3]
```diff
@@ -3,32 +3,95 @@
...
 
-    static func sharedConfiguration() -> Realm.Configuration {
-        var config = Realm.Configuration()
-        let directory = config.fileURL!.deletingLastPathComponent()
-        let url = directory.appendingPathComponent("shared.realm")
-        return Realm.Configuration(fileURL: url)
+    private static let fileManager = FileManager.default
+
+    private static let sharedRealmKey = "sharedRealmKey"
+    private static let waletsRealmKey = "waletsRealmKey"
```
[the remark this answers is not in the document]
Comment (OlegGordiichuk) [reply Trust/Core/Types/TrustRealmConfiguration.swift]: done

File: Trust/Core/Types/TrustRealmConfiguration.swift (hunk 2 above)
Comment (OlegGordiichuk) [reply Trust/Core/Types/TrustRealmConfiguration.swift]: Example from the Realm official doc.

let realmURL = Realm.Configuration.defaultConfiguration.fileURL!
let realmURLs = [
    realmURL,
    realmURL.appendingPathExtension("lock"),
    realmURL.appendingPathExtension("note"),
    realmURL.appendingPathExtension("management")
]
for URL in realmURLs {
    do {
        try FileManager.default.removeItem(at: URL)
    } catch {
        // handle error
    }
}

File: Trust/Core/Types/TrustRealmConfiguration.swift (hunk 1 above)
Comment (OlegGordiichuk) [reply Trust/Core/Types/TrustRealmConfiguration.swift]: will do

**Comment by vikmeup** [comment]
@OlegGordiichuk can you pull master?

File: Trust/EtherClient/PasswordGenerator.swift [hunk 4]
```diff
@@ -14,4 +14,13 @@ struct PasswordGenerator {
         let _ = SecRandomCopyBytes(kSecRandomDefault, bytesCount, &randomBytes)
         return randomBytes.map({ String(format: "%02hhx", $0) }).joined(separator: "")
     }
+
+    static func generateRandomData(bytesCount: Int) -> Data {
+        var keyData = Data(count: bytesCount)
+        var key = keyData
+        _ = key.withUnsafeMutableBytes {
+            SecRandomCopyBytes(kSecRandomDefault, keyData.count, $0)
```
[the remark this answers is not in the document]
Comment (OlegGordiichuk) [reply Trust/EtherClient/PasswordGenerator.swift]: ok I will use function above and convert string to data with utf8 encoding

[the remark this answers is not in the document]
Comment (OlegGordiichuk) [reply Trust/EtherClient/PasswordGenerator.swift]: It is a case when Realm required 64 bit data key. And with random hex string it gives a bit of the complexity to convert text string to 64 data and back.

**Comment by OlegGordiichuk** [comment]
@hewigovens yeah I fix issue with reinstall of the app and reuse of the key.

**Comment by vikmeup** [comment]
@OlegGordiichuk let's wait for a bit. Let me release latest version and then merge this.

**Comment by vikmeup** [comment]
@OlegGordiichuk can we make encrypted database only for new users? to simplify things and use the same key from the keychain(or if it's removed), that should be ok.

**Comment by vikmeup** [comment]
@OlegGordiichuk let's finish this one! 

Only encrypt newly created databases.

**Pull request closed without merging**